twilight-cache-inmemory/src/traits.rs

```diff
@@ -0,0 +1,247 @@
+//! Traits for implementing a [`InMemoryCache`] with custom structs.
+//!
+//! By default, the cache uses widely compatible default types that contain almost all
+//! fields that are present in the Discord API. Fields that are never used by the user
+//! will result in excess memory usage that will especially matter to big bots with a
+//! lot of cached data.
+//!
+//! The traits in this module allow creating custom cached representations of Discord
+//! API models compatible with the [`InMemoryCache`]. They may be mixed with the default
+//! types provided by twilight, which also implement these traits.
```

Comment:
It should be mentioned that these traits may break semver to keep up with new Discord models, for example, new requireed methods for new fields.